decaylanguage/decay/decay.py

```diff
@@ -0,0 +1,77 @@
+from collections import Counter
+
+
+class DaughtersDict(Counter):
+    """
+    Class holding a decay final state as a dictionary.
+    It is a building block for the digital representation of full decay chains.
+
+    Example
+    -------
+    A final state such as 'K+ K- K- pi+ pi0' is stored as
+    ``{'K+': 1, 'K-': 2, 'pi+': 1, 'pi0': 1}``.
+    """
+
+    __slots__ = ('_daughters')
+
+    def __init__(self, iterable=None, **kwds):
+        """
+        Default constructor.
+
+        Examples
+        --------
+        >>> # An empty final state
+        >>> dd = DaughtersDict()
+
+        >>> # Constructor from a dictionary
+        dd = DaughtersDict({'K+': 1, 'K-': 2, 'pi+': 1, 'pi0': 1})
+
+        >>> # Constructor from a list of particle names
+        >>> dd = DaughtersDict.from_list(['K+', 'K-', 'K-', 'pi+', 'pi0'])
+
+        >>> # Constructor from a string representing the final state
+        >>> dd = DaughtersDict('K+ K- pi0')
+        """
+        if iterable and isinstance(iterable, str):
+            iterable = iterable.split()
+        super(DaughtersDict, self).__init__(iterable, **kwds)
+
+    @classmethod
+    def from_list(cls, daughters):
```

Comment:
Is there a method of Counters that does this already?

Reply:
Indeed. Method hence removed. Thanks.